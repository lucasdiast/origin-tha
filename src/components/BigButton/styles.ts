import styled from 'styled-components';

export const Container = styled.div`
  max-width: 320px;
  border-radius: 32px;
  height: 40px;
  color: ${(props) => props.theme.colors.white};
  padding: 8px 0px;
  background-color: ${(props) => props.theme.colors.originBlue};
  line-height: 36px;
  margin: 0 20px;
  text-align: center;
`;
