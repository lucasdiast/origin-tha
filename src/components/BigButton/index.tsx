import React, { ReactElement, ReactChild } from 'react';
import { Container } from './styles';

interface AuxProps {
  children: ReactChild;
  onClickAction: () => void;
}

const BigButton = ({ children, onClickAction }: AuxProps): ReactElement => (
  <Container className="big-button" onClick={() => onClickAction()}>
    {children}
  </Container>
);

export default BigButton;
