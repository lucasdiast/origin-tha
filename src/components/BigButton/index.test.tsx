import { shallow } from 'enzyme';
import BigButton from './index';

describe('BigButton', () => {
  it('renders correctly', () => {
    const component = shallow(<BigButton>Text</BigButton>);

    expect(component).toMatchSnapshot();
  });
});
