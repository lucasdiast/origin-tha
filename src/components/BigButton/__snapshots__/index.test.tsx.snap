// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigButton renders correctly 1`] = `ShallowWrapper {}`;
